go/vt/vtgate/planbuilder/testdata/systemtables_cases.json

```diff
@@ -208,6 +190,30 @@
       "Original": "select * from information_schema.tables where table_schema = 'user' union select * from information_schema.tables where table_schema = 'main'",
```

Comment:
This reminds me that we could look into optimizing `UNION` queries to not _always_ split them up (e.g. in cases we will end up sending them to the same shard anyway).